packages/components/src/components/Calculator/useSavedCalculatorState.tsx

```diff
@@ -0,0 +1,56 @@
+import { useCallback, useMemo, useState } from "react";
+
+import { useViewerContext } from "../SquiggleViewer/ViewerProvider.js";
+import { SqCalculatorValueWithContext } from "./types.js";
+import { SqValueResult } from "./types.js";
+import { FormShape } from "./types.js";
+import { InputValues } from "./types.js";
+
+export type CalculatorState = {
+  hashString: string;
+  formValues?: FormShape;
+  inputValues?: InputValues;
+  fnValue?: SqValueResult;
+};
+
+// Takes a calculator value; returns the cache from ViewerContext for this calculator and a function for updating the cache.
+// Note that the returned cached value is never updated! It's write-only after the initial load.
```

Comment:
I'd change this last line to something like,
"The ``cachedState`` is only used for calculator initialization. After that, ``updatedCachedState`` gets called to keep the cache updated, but ``cachedState`` is no longer accessed."